hyok_configuration.go

```diff
@@ -0,0 +1,311 @@
+	GCPOIDCConfiguration   *GCPOIDCConfiguration
+	AzureOIDCConfiguration *AzureOIDCConfiguration
+	VaultOIDCConfiguration *VaultOIDCConfiguration
+}
+
+type KMSOptions struct {
+	// AWS
+	KeyRegion string `jsonapi:"attr,key-region,omitempty"`
+	// GCP
+	KeyLocation string `jsonapi:"attr,key-location,omitempty"`
+	KeyRingID   string `jsonapi:"attr,key-ring-id,omitempty"`
+}
+
+type HYOKConfiguration struct {
+	ID string `jsonapi:"primary,hyok-configurations"`
+
+	// Attributes
+	KEKID      string                  `jsonapi:"attr,kek-id"`
+	KMSOptions *KMSOptions             `jsonapi:"attr,kms-options,omitempty"`
+	Name       string                  `jsonapi:"attr,name"`
+	Primary    bool                    `jsonapi:"attr,primary"`
+	Status     HYOKConfigurationStatus `jsonapi:"attr,status"`
+	Error      *string                 `jsonapi:"attr,error"`
+
+	// Relationships
+	Organization      *Organization          `jsonapi:"relation,organization"`
+	OIDCConfiguration *OIDCConfigurationType `jsonapi:"polyrelation,oidc-configuration"`
+	AgentPool         *AgentPool             `jsonapi:"relation,agent-pool"`
+}
+
+type HYOKConfigurationsList struct {
+	*Pagination
+	Items []*HYOKConfiguration
+}
+
+type HYOKConfigurationsIncludeOpt string
+
+const (
+	HYOKConfigurationsIncludeHYOKCustomerKeyVersions HYOKConfigurationsIncludeOpt = "hyok_customer_key_versions"
+	HYOKConfigurationsIncludeOIDCConfiguration       HYOKConfigurationsIncludeOpt = "oidc_configuration"
+)
+
+type HYOKConfigurationsListOptions struct {
+	ListOptions
+	SearchQuery string                         `url:"q,omitempty"`
+	Include     []HYOKConfigurationsIncludeOpt `url:"include,omitempty"`
+}
+
+type HYOKConfigurationsCreateOptions struct {
+	ID string `jsonapi:"primary,hyok-configurations"`
+
+	// Attributes
+	KEKID      string      `jsonapi:"attr,kek-id"`
+	KMSOptions *KMSOptions `jsonapi:"attr,kms-options"`
+	Name       string      `jsonapi:"attr,name"`
+
+	// Relationships
+	Organization      *Organization          `jsonapi:"relation,organization"`
+	OIDCConfiguration *OIDCConfigurationType `jsonapi:"polyrelation,oidc-configuration"`
+	AgentPool         *AgentPool             `jsonapi:"relation,agent-pool"`
+}
+
+type HYOKConfigurationsReadOptions struct {
+	Include []HYOKConfigurationsIncludeOpt `url:"include,omitempty"`
+}
+
+type HYOKConfigurationsUpdateOptions struct {
+	ID string `jsonapi:"primary,hyok-configurations"`
+
+	// Attributes
+	KEKID      string      `jsonapi:"attr,kek-id"`
+	KMSOptions *KMSOptions `jsonapi:"attr,kms-options"`
+	Name       string      `jsonapi:"attr,name"`
+	Primary    bool        `jsonapi:"attr,primary"`
+
+	// Relationships
+	Organization      *Organization          `jsonapi:"relation,organization"`
+	OIDCConfiguration *OIDCConfigurationType `jsonapi:"polyrelation,oidc-configuration"`
+	AgentPool         *AgentPool             `jsonapi:"relation,agent-pool"`
+}
+
+func (h *HYOKConfigurationsListOptions) valid() error {
+	return nil
+}
```

Comment:
❓ do we need to specify an empty `valid`, or can we just not perform validation?